DNN Platform/Library/Common/Globals.cs

```diff
@@ -1,3842 +1,3927 @@
+                         portalSettings.PortalName + "</description>" + Environment.NewLine + "  <language>en-us</language>" + Environment.NewLine + "  <copyright>" +
+                         (!string.IsNullOrEmpty(portalSettings.FooterText) ? portalSettings.FooterText.Replace("[year]", DateTime.Now.Year.ToString()) : string.Empty) +
+                         "</copyright>" + Environment.NewLine + "  <webMaster>" + portalSettings.Email + "</webMaster>" + Environment.NewLine + strRSS + "   </channel>" + Environment.NewLine +
+                         "</rss>");
+                File.WriteAllText(fileName, strRSS.ToString());
+            }
+            else
+            {
+                if (File.Exists(fileName))
+                {
+                    File.Delete(fileName);
+                }
+            }
+        }
+
+        /// <summary>
+        /// injects the upload directory into raw HTML for src and background tags.
+        /// </summary>
+        /// <param name="strHTML">raw HTML text.</param>
+        /// <param name="strUploadDirectory">path of portal image directory.</param>
+        /// <returns>HTML with paths for images and background corrected.</returns>
+        public static string ManageUploadDirectory(string strHTML, string strUploadDirectory)
+        {
+            strHTML = ManageTokenUploadDirectory(strHTML, strUploadDirectory, "src");
+            return ManageTokenUploadDirectory(strHTML, strUploadDirectory, "background");
+        }
+
+        /// <summary>
+        /// injects the upload directory into raw HTML for a single token.
+        /// </summary>
+        /// <param name="strHTML">raw HTML text.</param>
+        /// <param name="strUploadDirectory">path of portal image directory.</param>
+        /// <param name="strToken">token to be replaced.</param>
+        /// <returns>HTML with paths for images and background corrected.</returns>
+        /// <remarks>
+        /// called by ManageUploadDirectory for each token.
+        /// </remarks>
+        public static string ManageTokenUploadDirectory(string strHTML, string strUploadDirectory, string strToken)
+        {
+            int p;
+            int r;
+            int s = 0;
+            int tLen;
+            string strURL;
+            var sbBuff = new StringBuilder(string.Empty);
+            if (!string.IsNullOrEmpty(strHTML))
+            {
+                tLen = strToken.Length + 2;
+                string uploadDirectory = strUploadDirectory.ToLowerInvariant();
+
+                // find position of first occurrance:
+                p = strHTML.IndexOf(strToken + "=\"", StringComparison.InvariantCultureIgnoreCase);
+                while (p != -1)
+                {
+                    sbBuff.Append(strHTML.Substring(s, p - s + tLen)); // keep charactes left of URL
+                    s = p + tLen; // save startpos of URL
```

Comment:
```suggestion
                    s = p + tLen; // save start position of URL
```

Reply:
Addressed in https://github.com/dnnsoftware/Dnn.Platform/pull/4215/commits/a83a2d5b2964f30fd677fcc2ffe519737a41d89c